populus/cli/main.py

```diff
@@ -26,50 +18,51 @@
 
 @click.group(context_settings=CONTEXT_SETTINGS)
 @click.option(
-    '--config',
-    '-c',
-    'config_file_path',
+    '--project-root-dir',
+    '-p',
+    'project_root_dir',
+    help=(
+        "Specify a populus project root directory"
+    ),
+    type=click.Path(exists=True),
+)
+@click.option(
+    '--global-config',
+    '-g',
+    'global_config_path',
+    help=(
+        "Specify a global populus configuration file, instead of the default"
+    ),
+    type=click.Path(exists=True),
+)
+@click.option(
+    '--warnings',
+    'show_warnings',
     help=(
-        "Specify a populus configuration file to be used.  No other "
-        "configuration files will be loaded"
+        "Show all warnings"
     ),
-    type=click.Path(exists=True, dir_okay=False),
+    default=False,
+    is_flag=True
 )
 @click.pass_context
-def main(ctx, config_file_path):
+def main(ctx, project_root_dir, show_warnings, global_config_path):
     """
     Populus
     """
     logger = get_logger_with_click_handler('populus')
+    if project_root_dir is None:
+        project_root_dir = os.getcwd()
+    ctx.obj = {}
+    ctx.obj['project_root_dir'] = project_root_dir
 
-    project = Project(config_file_path)
-
-    config_version = project.config['version']
-    subcommand_bypasses_config_version = ctx.invoked_subcommand in {'config', 'init'}
+    if show_warnings:
+        warnings.filterwarnings("always")
```

Comment:
Will you extract this change and get a pull request for it into master as a stand-alone improvement.